Reading List/Reading List/ReadingListTableViewController.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  ReadingListTableViewController.swift
+//  Reading List
+//
+//  Created by Dojo on 6/8/20.
+//  Copyright © 2020 Lambda School. All rights reserved.
+//
+
+import UIKit
+
+class ReadingListTableViewController: UITableViewController, BookTableViewDelegate {
+    
+    var bookController = BookController()
+    
+    private func bookFor(indexPath: IndexPath) -> Book {
+        if indexPath.section == 0 {
+            return bookController.readBooks[indexPath.row]
+        } else {
+            return bookController.unreadBooks[indexPath.row]
+        }
+    }
+    
+    func toggleHasBeenRead(for cell: BookTableViewCell) {
+        
+        if let book = cell.book {
+            bookController.updateRead(with: book)
+            tableView.reloadData()
+        }
+    }
+    
+    // MARK: - Table view data source
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        if section == 0 {
+            return bookController.readBooks.count
+        } else {
+            return bookController.unreadBooks.count
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "BookCell", for: indexPath) as? BookTableViewCell else {
+            return UITableViewCell()}
+        
+        let book = bookFor(indexPath: indexPath)
+        cell.book = book
+        cell.delegate = self
+        
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            let book = bookFor(indexPath: indexPath)
+            bookController.deleteBook(with: book)
+            tableView.deleteRows(at: [indexPath], with: .fade)
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        if section == 0 {
+            if bookController.readBooks.count > 0 {
+                return "Read Books"
+            } else {
+                return nil
+            }
+            
+        } else {
+            if bookController.unreadBooks.count > 0 {
+                return "Unread Books"
+            } else {
+                return nil
+            }
+        }
```

Comment:
Love, love, love that you made it not show the title for a section if the section has no books. Smart thinking.